### PR TITLE
ci: add GitHub Actions

### PR DESCRIPTION
**Pull request by dhmlau**
Signed-off-by: Diana Lau <dhmlau@ca.ibm.com>

Inspired by https://github.com/loopbackio/loopback-connector-mysql/pull/439, so I added @agnes512's authorship to this PR. 

Things to note in this PR:
- mysql version is 5.7. There are a few tests failed when using mysql 8.0 version. I can open a new issue to support mysql 8.0.  But that's a separate issue.
- added the commit linter and code linter as the original repo uses Jenkins which no longer works

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+
+env:
+  CI: true
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
```
Comment (marioestradarosa) [inline .github/workflows/ci.yml:14]: Is it possible to check on ubuntu-20.x ?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        node-version: [12, 14, 16]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ankane/setup-mysql@v1
+        with:
+          mysql-version: 5.7
```
Comment (marioestradarosa) [inline .github/workflows/ci.yml:27]: I guess this relates tu ubuntu 18.04 right?, because  I have seen version 8.0 being adopted.

**Review by marioestradarosa — COMMENTED** [review]
Nice @dhmlau

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      - run: |
+          sudo mysql -e "CREATE USER 'test'@'localhost' IDENTIFIED BY 'test'"
+          sudo mysql -u root -e "GRANT ALL PRIVILEGES ON *.* TO 'test'@'localhost'"
+          sudo mysql -u root -e "ALTER USER 'test'@'localhost' IDENTIFIED WITH mysql_native_password by 'test'"
+          sudo mysql -u root -e "FLUSH PRIVILEGES"
+      - run: npm install
+      - run: npm test
+        env:
+          MYSQL_USER: test
```
Comment (marioestradarosa) [inline .github/workflows/ci.yml:36]: You can use gitHub Secrets {{ }} here instead of the test/test.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dhmlau) [reply .github/workflows/ci.yml:27]: yes. It seems like our current code only works with mysql 5.7 and there're some issues with `ubuntu-latest` (i.e. ubuntu-20.04) + mysql 5.7 combination. This is one of the threads I've read: https://ovirium.com/blog/how-to-make-mysql-work-in-your-github-actions/.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dhmlau) [reply .github/workflows/ci.yml:14]: pls see comment below.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dhmlau) [reply .github/workflows/ci.yml:36]: let me set it up. thanks for the suggestion.

**Review by marioestradarosa — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      matrix:
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+        node-version: [12, 14, 16]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ankane/setup-mysql@v1
```
Comment (agnes512) [inline .github/workflows/ci.yml:25]: Glad to see this setup works! I wasn't able to get MySQL 5.7 to run properly with other setups 👍

**Review by agnes512 — APPROVED** [review]
Thanks for adding it!

**Pull request merged**